src/components/general/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import ErrorMessage, { ErrorTypes } from "../ErrorMessage";
+import PropTypes from "prop-types";
+
+type ErrorBoundaryProps = {
+    hasError?: boolean,
+    errorType: ErrorTypes,
+    message?: string,
+};
+
+export default class ErrorBoundary extends React.Component<ErrorBoundaryProps> {
+    static propTypes = {
+        hasError: PropTypes.bool,
+        onRetry: PropTypes.func,
+        errorType: PropTypes.oneOf(Object.values(ErrorTypes)),
+        message: PropTypes.node,
+        maxRetries: PropTypes.number,
+    };
+    static defaultProps = {
+        errorType: ErrorTypes.error
+      }
+
+    state = {
+        hasError: false,
+        retries: 0,
+    };
+
+    componentDidCatch(error, errorInfo) {
+
+        this.setState({ hasError: true });
```

Comment:
Persist the error and errorInfo to provide more info to the error message?

Reply:
Now uses the error.message to provide more info. 
Is it necessary to provide more info than the message to the user?
All error information was used to provide to appInsight, but we don't have a implementation for that yet.

Reply:
Added error and errorMessage on state for now